core/src/banking_stage/scheduler_messages.rs

```diff
@@ -0,0 +1,54 @@
+use {
+    crate::immutable_deserialized_packet::ImmutableDeserializedPacket,
+    solana_poh::poh_recorder::Slot, solana_sdk::transaction::SanitizedTransaction, std::sync::Arc,
+};
+
+/// A unique identifier for a transaction batch.
+#[derive(Clone, Copy, Debug, Hash, PartialEq, Eq)]
+pub struct TransactionBatchId(u64);
+
+impl TransactionBatchId {
+    pub fn new(index: u64) -> Self {
+        Self(index)
+    }
+}
+
+/// A unique identifier for a transaction.
+#[derive(Clone, Copy, Debug, Hash, PartialEq, Eq)]
+pub struct TransactionId(u64);
+
+impl TransactionId {
+    pub fn new(index: u64) -> Self {
+        Self(index)
+    }
+}
+
+/// Message: [Scheduler -> Worker]
+/// Transactions to be consumed (executed, recorded, committed)
+pub struct ConsumeWork {
+    pub batch_id: TransactionBatchId,
+    pub transaction_ids: Vec<TransactionId>,
```

Comment:
naming: `ids` to be consistent with `ForwardWork`? (a field with identical type `Vec<TransactionId>` isn't called as `packet_ids` there currently).

Reply:
done: 005ebd9edf77568f9a8e28ab1b1d7e17a17649bd

I originally had `ids`, then moved away to clarify from `batch_id` (which I had originally named `id`...then never changed these back! I think `ids` is already clear enough, so changed to be consistent between the messages.